test/PidTest.cpp

```diff
@@ -0,0 +1,151 @@
+/**
+ *  @file PidTest.cpp
+ *  @brief PidGainクラス,Pidクラスをテストする
+ *  @author Hisataka-Hagiyama,uchyam
+ */
+
+#include "Pid.h"
+#include <gtest/gtest.h>
+
+namespace etrobocon2021_test {
+  TEST(PidGainTest, gain)
+  {
+    double expected_p = 0.1;
+    double expected_i = 0.03;
+    double expected_d = 0.612;
+    PidGain actualPidGain(0.1, 0.03, 0.612);
+    EXPECT_DOUBLE_EQ(expected_p, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_i, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_d, actualPidGain.kd);
+  }
+
+  TEST(PidGainTest, gain_zero)
+  {
+    double expected_p = 0.0;
+    double expected_i = 0.0;
+    double expected_d = 0.0;
+    PidGain actualPidGain(0.0, 0.0, 0.0);
+    EXPECT_DOUBLE_EQ(expected_p, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_i, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_d, actualPidGain.kd);
+  }
+
+  TEST(PidGainTest, gain_minus)
+  {
+    double expected_p = -0.5;
+    double expected_i = -0.2;
+    double expected_d = -0.3;
+    PidGain actualPidGain(-0.5, -0.2, -0.3);
+    EXPECT_DOUBLE_EQ(expected_p, actualPidGain.kp);
+    EXPECT_DOUBLE_EQ(expected_i, actualPidGain.ki);
+    EXPECT_DOUBLE_EQ(expected_d, actualPidGain.kd);
+  }
+
+  TEST(PidTest, caluculatePid)
+  {
+    static const double DELTA = 0.01;
+    double expected_p = 0.6;
+    double expected_i = 0.02;
+    double expected_d = 0.03;
+    Pid actualPid(0.6, 0.02, 0.03, 70);

```

Comment:
個人的には、以下のように書いたほうが分かりやすい(?)と思ったのですが、以下のように書かなかった理由などはあるのでしょうか？

現状のコードの書き方のほうが一般的とか？

```suggestion
    PidGain actualPidGain(expected_p, expected_i, expected_d);
```

Reply:
意図してこの書き方をしたという訳ではありません。
確かに挙げられた書き方の方が簡潔だと思います。
修正しておきます。